arch/xtensa/src/esp32s3/Kconfig

```
@@ -517,6 +533,72 @@ config ESP32S3_GPIO_IRQ
 	---help---
 		Enable support for interrupting GPIO pins.
 
+menu "SPI configuration"
+	depends on ESP32S3_SPI
+
+config ESP32S3_SPI_SWCS
+	bool "SPI software CS"
+	default n
+	---help---
+		Use SPI software CS.
+
+config ESP32S3_SPI_UDCS
+	bool "User defined CS"
+	default n
+	depends on ESP32S3_SPI_SWCS
+	---help---
+		Use user-defined CS.
+
+if ESP32S3_SPI2
+
+config ESP32S3_SPI2_CSPIN
+	int "SPI2 CS Pin"
+	default 10
+	range 0 21
+
+config ESP32S3_SPI2_CLKPIN
+	int "SPI2 CLK Pin"
+	default 6
+	range 0 21
+
+config ESP32S3_SPI2_MOSIPIN
+	int "SPI2 MOSI Pin"
+	default 7
+	range 0 21
```

```suggestion
	range 0 48
```
ESP32-S3 pin numbers range from 0 to 48.